### DISCUSSION TITLE
docs: removal of the `--save-dev` flag

### DISCUSSION THREAD

**Discussion by THETCR** [amilajack/eslint-plugin-compat#543] [General]
The removal of the `npm-install` flag for saving this package to the development dependencies in the `README.md` seems like a questionable choice.
The package is meant to be used as support for development. People who will blindly copy the `npm install eslint-plugin-compat` command will now install it as a production dependency, while the majority should install it as a development dependency.

What was the reasoning behind this?

**Comment by amilajack** [comment]
the primary consumer of eslint-plugin-compat is apps. the distinction between devDeps and devs for apps isn't very valuable so frameworks like next have been putting build dependencies in deps.

but if palette's primary consumer was libs, this would be a different story

**Comment by THETCR** [comment]
@amilajack 
Even though it won't be served to the client, it will still cause the dependency to be unnecessarily installed on the server.
Which is a waste of resources. Think about data being transferred over the internet, I/O operations and NodeJS traversing the files when trying to resolve something. With CI/CD this would mean constantly wasting that bandwidth.

Only Next.js does this in their generator, but that doesn't mean it's a good idea.
The only use case were this wouldn't have any downsides is when serving the app as a distribution bundle through, for instance, a CDN.

**Comment by amilajack** [comment]
good points, here're my observations

> With CI/CD this would mean constantly wasting that bandwidth

linting is often run in ci/cd. 

> Only Next.js does this in their generator, but that doesn't mean it's a good idea.

create-react-app also seems to do this as well:

```json
  "dependencies": {
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },
```

**Comment by amilajack** [comment]
closing but let's continue the discussion
